File: Python/002.add two number.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def addTwoNumbers(self, l1, l2):
        result = ListNode(0)
        current,carry = result,0
        while l1 != None or l2 != None:
            res = carry
            if l1 != None:
                res += l1.val
                l1 = l1.next
            if l2 != None:
                res += l2.val
                l2 = l2.next
            carry,res = res // 10,res % 10
            current.next = ListNobe(res)
            current = current.next
        if carry == 1:
            current.next = ListNobe(1)
        return result.next

File: Javascript/002.add two number.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */
/**
 * @param {ListNode} l1
 * @param {ListNode} l2
 * @return {ListNode}
 */
var addTwoNumbers = function(l1, l2) {
    var result = new ListNobe(0);
    var current = result;
    var carry = 0;
    while (l1 || l2){
        if (l1){
            var q = l1.val;
            l1 = l1.next;
        }else {
            q =0;
        }
        if (l2){
            var p = l2.val;
            l2 = l2.next
        }else {
            p = 0
        }
        var res = carry + q + p;
        if(res > 9){
            carry = 1;
            res = res % 10;
        }else {
            carry = 0;
        }
        current.next = new ListNobe(res);
        current = current.next;
    }
    if (carry == 1){
        current.next = new ListNobe(1);
    }
    return result.next;
};
